### ISSUE TITLE
minor installation issue on linux machine

### ISSUE THREAD

**Issue by mfastudillo** [USEPA/LCIAformatter#59]
`pip install git+https://github.com/USEPA/LCIAformatter.git@joss_review#egg=lciafmt` throws an error when trying to install on a ubuntu machine. The problem seems to be that `pyodbc` would not install with  `pip install pyodbc` ... see [[pyodbc install instructions](https://github.com/mkleehammer/pyodbc/wiki/Install)]. I therefore installed first `unixodbc-dev` and then  `pip install git+https://github.com/USEPA/LCIAformatter.git@joss_review#egg=lciafmt` run apparently successfully. However if I try to import lciafmt from a ipython session I get the following error:

fatal: not a git repository (or any of the parent directories): .git

I am not sure how this problem can be handled, but I guess it can be at least noted on the installation instructions.  

openjournals/joss-reviews#3392

**Comment by tngTUDOR** [comment]
My flow to install with Linux is as follows:

```
conda create -y -n lciafmt python=3.9
conda activate lciafmt
conda install pyodbc
pip install git+https://github.com/USEPA/LCIAformatter.git@joss_review#egg=lciafmt
```

And this install will work fine. 
However, pyodbc installed on linux like this will not work. See below for the error message. 

```
>>> lciafmt.clear_cache()
INFO delete cache folder /tmp/lciafmt
>>> impact_m = lciafmt.get_method(lciafmt.Method.ImpactWorld)
INFO get method ImpactWorld+
WARNING Please install drivers to remotely connect to Access Database. Drivers only available on windows platform.For instructions visit: https://github.com/mkleehammer/pyodbc/wiki/Connecting-to-Microsoft-Access
INFO download from https://www.dropbox.com/sh/2sdgbqf08yn91bc/AABIGLlb_OwfNy6oMMDZNrm0a/IWplus_public_v1.3.accdb?dl=1 to /tmp/lciafmt/Impact_World.accdb
INFO read ImpactWorld+ from file /tmp/lciafmt/Impact_World.accdb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/maynardjk/workspaces/joss/LCIAformatter/lciafmt/__init__.py", line 81, in get_method
    return impactworld.get(file=file, url=url)
  File "/home/maynardjk/workspaces/joss/LCIAformatter/lciafmt/iw.py", line 36, in get
    df = _read(f)
  File "/home/maynardjk/workspaces/joss/LCIAformatter/lciafmt/iw.py", line 61, in _read
    cnxn = pyodbc.connect(connStr)

pyodbc.Error: ('01000', "[01000] [unixODBC][Driver Manager]Can't open lib 'Microsoft Access Driver (*.mdb, *.accdb)' : file not found (0) (SQLDriverConnect)")

```

**Comment by bl-young** [comment]
@mfastudillo 

> fatal: not a git repository (or any of the parent directories): .git

This error occurs when you are not connected to a git - from this [block](https://github.com/USEPA/LCIAformatter/blob/master/lciafmt/util.py#L40-L44). Despite it being described as a fatal error, it has limited impact except that the git-hash won't be [appended to the file name](https://github.com/USEPA/esupy/wiki/File-Naming-Structure) of any method you produce. You should be able to proceed with all functions. It would be good to find a way to avoid that error, or at least explain it as such in the readme.

**Comment by bl-young** [comment]
@tngTUDOR 
> WARNING Please install drivers to remotely connect to Access Database. Drivers only available on windows platform. For instructions visit: https://github.com/mkleehammer/pyodbc/wiki/Connecting-to-Microsoft-Access

We have added this warning to point users to this explanation for how to download the correct driver for pyodbc if you don't have it installed. However it seems this may not be feasible in linux. For ImpactWorld+, it might be limited to Windows only.

**Comment by WesIngwersen** [comment]
What about making pyodbc an optional dependency in the setup.py, like
[reference in setup tools docs](https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html)

``` 
setup(
...
extras_require={
        "ImpactWorld+":  ["pyodbc>=4.0.30;platform_system='Windows'"],
    }
}
```
As I interpret the docs, this setup will not install pyodbc by default with pip install.  You would have to use 
>pip install LCIAformatter["ImpactWorld+"] 

to install it, and still then it would only install pyodbc if it is a Windows machine due to the statement of that dependency afterwards.

**Comment by WesIngwersen** [comment]
Following that to also make pyodbc behave like an optional requirements in respect to `requirements.txt` we could move that line into an `ImpactWorld_requirements.txt` as pip permits install with multiple requirements files

**Comment by WesIngwersen** [comment]
@mfastudillo @tngTUDOR  I have implemented the changes I suggested in my comments. Can one of you test that it works on Linux and will not install pyodbc?  I hope this solution is acceptable to you given the constraints of the Impact World+ data being published as an Access db for this version. We can communicate with that team to note this limitation.

**Comment by tngTUDOR** [comment]
I agree  @WesIngwersen, it would be better for Open Source Initiatives if ImpactW+ was published 
- in a more widespread db format (along MS access ) or at least something supporting Mac OS, Linux and Windows without too much trouble. Sqlite is the only thing coming to mind right now, given the context, but this depends on the features borrowed from Access. Of course, I would prefer feather or plain csv as last resource.
- definitely not in dropbox ! 

You have all my support to ask the IW+ team about this two things. Maybe they can use lciaformatter with a windows machine to directly publish a json-ld version on github

**Comment by mfastudillo** [comment]
Hello

I tried doing this on a linux machine : 

`python3 -m venv eparev5`
`source eparev5/bin/activate`
`pip install git+https://github.com/USEPA/LCIAformatter.git@v1.0.0#egg=lciafmt`

and it ends up not using the wheel because it needs to be installed first. it does install pyodbc. If I install the wheels package before LCIAformatter, it also installs pyodbc. I'll try to restart everything from scratch

![image](https://user-images.githubusercontent.com/11621547/126895459-523780a0-f640-4696-a832-816e8af83082.png)

**Comment by mfastudillo** [comment]
I tried 

```
python3 -m venv eparev6
source eparev6/bin/activate
pip install git+https://github.com/USEPA/LCIAformatter.git@joss_review#egg=lciafmt
```
and it does install lciafmt without installing pyodbc (although it complains of wheel not being installed). But if I try to import `lciafmt` I run into the same problem. 

![image](https://user-images.githubusercontent.com/11621547/126905569-0b1f8ae1-6736-4b97-b45d-9389fbf62af3.png)

Nonetheless I can import the methods e.g. `lciafmt import get_method` and they seem to work fine

**Comment by WesIngwersen** [comment]
@mfastudillo  That error message is from trying to run a git command, so it seems `import lciafmt` is indirectly calling git. I'm guessing that is [here when we try to get a git hash for the current module](https://github.com/USEPA/LCIAformatter/blob/9c225425f6f62a47cfd7cf9f6f05f987db6d6ed8/lciafmt/util.py#L41), but it's inside a try clause, so I'm not sure why its returning that error. This is probably a separate issue.

**Comment by bl-young** [comment]
The git hash issue in the most recent comment has been resolved (#60). 

I believe this issue can also be closed with the updated install procedure for linux.
